apps/server/src/shared/domain/entity/submission.entity.ts

```diff
@@ -1,12 +1,12 @@
-import { Entity, ManyToOne, Collection, Property, ManyToMany, Index } from '@mikro-orm/core';
+import { Collection, Entity, Index, ManyToMany, ManyToOne, Property } from '@mikro-orm/core';
 
 import { EntityId } from '../types';
 
 import { BaseEntityWithTimestamps } from './base.entity';
 import type { CourseGroup } from './coursegroup.entity';
-import type { User } from './user.entity';
 import type { File } from './file.entity';
 import type { Task } from './task.entity';
+import type { User } from './user.entity';
 
```

Comment:
Bleibt `task: Task;` weiterhin mandatory?

Reply:
Wie meinst du "weiterhin"? Was hat sich in der Hinsicht geändert?

Ich würde allgemein sagen, eine submission ohne task macht keinen Sinn.

Reply:
Ich bin an dieser Stelle selber etwas unsicher mit MikroOrm. Was passiert denn mit der Property, wenn weder ein Populate ausgeführt, noch eine Loading Strategy in der Entity definiert wird? Steht dann die TaskId oder die Task in der Property?

Reply:
Wenn nicht populiert, steht die id in der property.

Reply:
Und das die anderen Props undefined sind, ist in der Submission Klasse nicht abgebildet und dadurch erkennt Typescript auch nicht, dass diese Properties undefined sein können.
Um es korrekt in der Entity Klasse abzubilden, habe ich in der MikroORM Doku den Reference Wrapper gefunden. Damit würde die Task Property dann wahrscheinlich so aussehen:
```ts
@ManyToOne('Task', { fieldName: 'homeworkId' })
@Index()
task: ref<Task>;
```
@CeEv @SevenWaysDP Warum verwenden wir den Reference Wrapper nicht für die Reference Properties? Das betrifft ja wahrscheinlich auch viele andere Fälle.